src/battle-tooltips.ts

```diff
@@ -795,12 +795,19 @@ class BattleTooltips {
 				}
 			}
 
-			let types = this.getPokemonTypes(pokemon);
+			let types = serverPokemon?.terastallized ? [serverPokemon.teraType] : this.getPokemonTypes(pokemon);
 
-			if (clientPokemon && (clientPokemon.volatiles.typechange || clientPokemon.volatiles.typeadd)) {
+			if (pokemon.terastallized) {
+				text += `<small>(Terastallized)</small><br />`;
+			} else if (clientPokemon?.volatiles.typechange || clientPokemon?.volatiles.typeadd) {
 				text += `<small>(Type changed)</small><br />`;
 			}
 			text += types.map(type => Dex.getTypeIcon(type)).join(' ');
+			if (pokemon.terastallized) {
+				text += ` <small>(base: ${this.getPokemonTypes(pokemon, true).map(type => Dex.getTypeIcon(type)).join(' ')})</small>`;
+			} else if (serverPokemon) {
```

Comment:
`serverPokemon?.teraType`? (I'm assuming here you're only sending the Tera type for Gen 9 battles.)